llvm/lib/Analysis/ConstantFolding.cpp

```diff
@@ -2754,27 +2754,28 @@ static Constant *ConstantFoldIntrinsicCall2(Intrinsic::ID IntrinsicID, Type *Ty,
           ((Mask & fcPosInf) && Op1V.isPosInfinity());
         return ConstantInt::get(Ty, Result);
       }
+      case Intrinsic::powi: {
+        int exp = static_cast<int>(Op2C->getZExtValue());
```

Comment:
```suggestion
        int Exp = static_cast<int>(Op2C->getZExtValue());
```
Though while you're here, maybe fix this as well? Surely this should be getSExtValue(), and we shouldn't cast away the top bits either?